library/x509_create.c

```diff
@@ -382,6 +382,10 @@ int mbedtls_x509_set_extension(mbedtls_asn1_named_data **head, const char *oid,
 {
     mbedtls_asn1_named_data *cur;
 
+    if (0xFFFFFFFF == (uint32_t) val_len) {
```

Comment:
A better (platform-independent) way to do this would be:
```suggestion
    if (val_len > (SIZE_MAX  - 1)) {
```

Reply:
Yes, you are right!